packages/contracts/src/actions/typegen/buildTypes.ts

```diff
@@ -0,0 +1,17 @@
+import { runTypegen } from '@fuel-ts/abi-typegen';
+
+import { getArtifactPaths } from '../../helpers/sway';
+import type { ContractsConfig } from '../../types';
+
+// Generate types using typechain
+// and typechain-target-fuels modules
+export async function buildTypes(config: ContractsConfig) {
+  const cwd = process.cwd();
+  // find all files matching the glob
+  const filepaths = getArtifactPaths(config.contracts);
+  await runTypegen({
+    cwd,
+    filepaths,
+    output: config.types.output,
+  });
+}
```

Comment:
```suggestion
import { spawn } from 'child_process';
import type { Config } from 'src/types';
// Generate types
export async function buildTypes(config: Config) {
  return new Promise((resolve, reject) => {
    const typeGeneration = spawn('pnpm', [
      'exec',
      'fuels',
      'typegen',
      '-i',
      config.types.artifacts,
      '-o',
      config.types.output,
    ]);
    typeGeneration.on('exit', (code) => {
      if (!code) return resolve(code);
      typeGeneration.kill();
      reject();
    });
  });
}
```
We will also need to add `fuels` as a dependency from wherever the contracts package is run from.